src/start/RealmScreen.js

```diff
@@ -76,7 +77,13 @@ class RealmScreen extends PureComponent<Props, State> {
     const { progress, error, realm } = this.state;
 
     return (
-      <Screen title="Welcome" padding centerContent keyboardShouldPersistTaps="always">
+      <Screen
+        title="Welcome"
+        canGoBack={!this.props.navigation.state.params?.initial}
```

Comment:
This is what `dangerouslyGetParent` is documented to be for.

It's still a hack, mind (as if the name didn't give it away). The "right" way is probably to use a react-navigation `AppContainer`, rather than trying to generalize this somehow.

Reply:
Hmm fascinating! I'll take a look.

Reply:
Hmm, so here's the docs:
https://reactnavigation.org/docs/en/2.x/navigation-prop.html#dangerouslygetparent-get-parent-navigator

It looks like this will get us the `StackNavigator` object we call `AppNavigator`. And from there, we'd take its `routes` property or something like that?

I think that's the same purpose we currently serve with the `nav` reducer and selectors -- note `getNavigationRoutes` in particular.

One reason we moved away from that is that it didn't have the right behavior on transitions. You'd be on one screen, and you'd hit a button that navigated to another... and for a few hundred ms, the old screen is still visible (at least in part), but the *overall* nav state has already changed, so if the screen uses the overall nav state to decide how to render then we get glitches. The `props.navigation.state.params` value, on the other hand, remains at the value that belonged to *this* screen.

Reply:
On `AppContainer`: reading up on that, it looks like it's part of what we already have (it's implicit in `createStackNavigator` up through 2.x, and that's what we use in `AppNavigator.js`); but the part of it that handles showing the app bar and back-navigation button, we turn off by passing `headerMode: 'none'`.

Seems quite likely that an eventual good answer here would look like providing our own implementation of the app bar, integrated with the navigation framework, rather than disabling it. Docs for that appear to be
https://reactnavigation.org/docs/en/2.x/navigation-views.html
https://reactnavigation.org/docs/en/navigation-views.html
and in particular the latter links to
https://github.com/react-navigation/stack/blob/1.0/src/views/Header/Header.tsx

I'll augment the comment to mention that.